backend/config/routes.rb

```diff
@@ -3,7 +3,7 @@
 
   namespace :admin do
     resources :users
-
+    get '/teachers', action: :teacher, controller: :users
```

Comment:
having custom routes is not a good signal for simple operations, try to use this https://idratherbewriting.com/learnapidoc/docapis_doc_parameters.html#query_string_parameters